process/sync/baseSync.go

```diff
@@ -103,207 +102,8 @@ type baseBootstrap struct {
-	}
-
-	return nil
-}
-
-func (boot *baseBootstrap) computeHighestNonce(hdrNonceHashDataUnit dataRetriever.UnitType) uint64 {
-	highestNonceInStorer := uint64(0)
-
-	for {
-		highestNonceInStorer++
-		nonceToByteSlice := boot.uint64Converter.ToByteSlice(highestNonceInStorer)
-		err := boot.store.Has(hdrNonceHashDataUnit, nonceToByteSlice)
-		if err != nil {
-			highestNonceInStorer--
-			break
-		}
-	}
-
-	return highestNonceInStorer
-}
-
-func (boot *baseBootstrap) applyBlock(shardId uint32, nonce uint64) error {
-	header, headerHash, err := boot.storageBootstrapper.getHeader(shardId, nonce)
-	if err != nil {
-		return err
-	}
-
-	log.Debug("apply block",
-		"nonce", header.GetNonce(),
-		"round", header.GetRound(),
-	)
-
-	blockBody, err := boot.storageBootstrapper.getBlockBody(header)
-	if err != nil {
-		return err
-	}
-
-	err = boot.blkc.SetCurrentBlockBody(blockBody)
-	if err != nil {
-		return err
-	}
-
-	err = boot.blkc.SetCurrentBlockHeader(header)
-	if err != nil {
-		return err
-	}
-
-	boot.blkc.SetCurrentBlockHeaderHash(headerHash)
-
-	return nil
-}
-
-func (boot *baseBootstrap) cleanupStorage(
-	nonce uint64,
-	blockUnit dataRetriever.UnitType,
-	hdrNonceHashDataUnit dataRetriever.UnitType,
-) {
-	errNotCritical := boot.storageBootstrapper.removeBlockBody(nonce, blockUnit, hdrNonceHashDataUnit)
-	if errNotCritical != nil {
-		log.Debug("remove block body",
-			"nonce", nonce,
-			"error", errNotCritical.Error(),
-		)
-	}
-
-	errNotCritical = boot.removeBlockHeader(nonce, blockUnit, hdrNonceHashDataUnit)
-	if errNotCritical != nil {
-		log.Debug("remove block header",
-			"nonce", nonce,
-			"error", errNotCritical.Error(),
-		)
-	}
-}
-
-func (boot *baseBootstrap) removeBlockHeader(
-	nonce uint64,
-	blockUnit dataRetriever.UnitType,
-	hdrNonceHashDataUnit dataRetriever.UnitType,
-) error {
-	return nil
-}
-
-func (boot *baseBootstrap) getShardHeaderFromStorage(
-	shardId uint32,
-	nonce uint64,
-) (data.HeaderHandler, []byte, error) {
-
-	nonceToByteSlice := boot.uint64Converter.ToByteSlice(nonce)
-	hdrNonceHashDataUnit := dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(shardId)
-	headerHash, err := boot.store.Get(hdrNonceHashDataUnit, nonceToByteSlice)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	header, err := process.GetShardHeaderFromStorage(headerHash, boot.marshalizer, boot.store)
-
-	return header, headerHash, err
-}
-
-func (boot *baseBootstrap) getMetaHeaderFromStorage(
-	shardId uint32,
-	nonce uint64,
-) (data.HeaderHandler, []byte, error) {
-
-	nonceToByteSlice := boot.uint64Converter.ToByteSlice(nonce)
-	headerHash, err := boot.store.Get(dataRetriever.MetaHdrNonceHashDataUnit, nonceToByteSlice)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	header, err := process.GetMetaHeaderFromStorage(headerHash, boot.marshalizer, boot.store)
-
-	return header, headerHash, err
+	bootStorer           process.BootStorer
+	storageBootstrapper  process.BootstrapperFromStorage
```

Comment:
Isn't bootStorer linked with storageBootstrapper? If so, it can expose the needed storer actions instead of having two different components everywhere

Reply:
We need 2 different component because on block processor we need just bootStorer and in base sync we need both components.